src/mon/ConfigKeyService.cc

```diff
@@ -109,6 +109,17 @@ bool ConfigKeyService::store_has_prefix(const string &prefix)
   return false;
 }
 
+static bool is_binary_string(const string& s)
+{
+  for (auto c : s) {
+    // \n and \t are escaped in JSON; other control characters are not.
+    if ((c < 0x20 && c != '\n' && c != '\t') || c >= 0x7f) {
```

Comment:
In ceph_argparse.py, we restrict inputs to ASCII.  I'm not sure we have any ASCII restriction on the outputs though?  If not, we should probably consider UTF-8 content to be a string rather than binary

Reply:
I questioned "only \t and \n", so went and read https://tools.ietf.org/html/rfc8259#page-9, which seems to define an escaping for any value?...  Maybe we should do that instead (unless it's ridiculously long or something)?  It's not inconceivable that that could be useful.

If we don't want to go full \uNNNN, then it seems like at least we ought to do the special-two-chars from the above section.

Reply:
(and apologies for not addressing this in the original addition of "dump".)

Reply:
The escaping code is here (https://github.com/ceph/ceph/blob/master/src/common/escape.cc#L176) and it's pretty simplistic.  We can probably do better, but I'm not sure how much it matters.  IIUC that code assumes (and is sufficient for) utf8, but I confess I don't really understand unicode.  Even if we make it escape more stuff, there are still (binary) strings we can't escape in JSON (that aren't valid unicode strings... e.g., "\0"), so I'm not sure how much it matters.

Reply:
I had been thinking that for UTF-8 strings we would just pass them through in UTF-8 output, but I suspect we've never really vetted that whole output path or defined its encoding, so maybe it is safer to just throw out everything non-ASCII